### ISSUE TITLE
Worker standalone process?

### ISSUE THREAD

**Issue by justinmchase** [Azure/azure-functions-dotnet-worker#839]
Right now if I run `func host start` I'm not sure what its doing but it starts up my functions and all works.

But my project is producing an exe `<OutputType>Exe</OutputType>` and if I just run it with `dotnet run` it blows up with an error about ` The gRPC channel URI 'http://:0' could not be parsed.`

I'm just wondering, is it possible to just configure my process to simply run as if I was doing `func host start`?

**Comment by justinmchase** [comment]
May be related to #442

**Comment by justinmchase** [comment]
How can I configure the gRPC channel here?

https://github.com/Azure/azure-functions-dotnet-worker/blob/704de76b5693bc24bdf6a9836ba8a109517c6a41/src/DotNetWorker.Grpc/GrpcServiceCollectionExtensions.cs#L58

**Comment by justinmchase** [comment]
So am I to understand that `func host start` is essentially the main process and then it is creating multiple child processes out of my exe and passing in some CLI args for IPC? So I always have to run it via the func host?

**Comment by kshyju** [comment]
The C# out-of-process function app cannot run by itself. It runs **_with the function host._**

On a high level, an out of process/isolated function app runs using two pieces, one is the function host, and the other is the isolated function application. When you run `func host start`, the host process will startup and then will internally invoke the out-of-process function app. When it does this, the host passes some information the worker needs, as arguments, so that the worker can use those to properly communicate with the host (things like the port number to be used for GRPC communication between host and worker) 

When you run `dotnet start` on the isolated app, the host process is not involved at all. The isolated process startup code expects the above-mentioned arguments (such as port number of host), hence failing with the exception you are seeing.

In short, you should not run `dotnet start` on an isolated process. You should always run `func start`

Hope this helps. Let us know if you still have questions.
